drivers/adc/adc_stm32.c

```diff
@@ -275,6 +147,8 @@ struct adc_stm32_cfg {
 	int8_t temp_channel;
 	int8_t vref_channel;
 	int8_t vbat_channel;
+	const uint32_t sampling_time_table[STM32_NB_SAMPLING_TIME];
```

Comment:
It looked to me like the largest existing sampling time value was 814, which would easily fit into a `uint16_t`.  

Due to alignment, there will be a padding byte between the three `uint8_t` fields before `sampling_time_table`.  Moving these fields up in the struct, i.e. the order `pcfg`, `sampling_time_table`, `sampling_time_common_channel_nb`, `temp_channel`, would remove the need for padding.  In other words, sort in descending order of alignment.

Reply:
Done